### PR TITLE
moveit: Add packages from MoveIt! project

### PR DESCRIPTION
**Pull request by rojkov**
Particularly this commit brings in

* moveit_core;
* moveit_ros_perception;
* moveit_ros_planning;
* moveit_msgs

and their build dependencies like

* object_recognition_msgs;
* srdfdom.

Also they are included into packagegroup-ros-world.bb.

### CODE REVIEW

File: recipes-ros/moveit/moveit-core_0.7.8.bb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+DESCRIPTION = "Core libraries used by MoveIt!"
+SECTION = "devel"
+LICENSE = "BSD"
+LIC_FILES_CHKSUM = "file://package.xml;beginline=14;endline=14;md5=d566ef916e9dedc494f5f793a6690ba5"
+
+DEPENDS = "roslib boost eigen-conversions fcl octomap eigen-stl-containers geometric-shapes kdl-parser moveit-msgs srdfdom cmake-modules"
+
+require moveit-${PV}.inc
+
+SRC_URI = "https://github.com/ros-planning/${ROS_SPN}/archive/${PV}.tar.gz;downloadfilename=${ROS_SP}.tar.gz"
```
Comment (bulwahn) [inline recipes-ros/moveit/moveit-core_0.7.8.bb]: Drop this line; it is already part of the .inc file.

File: recipes-ros/moveit-msgs/moveit-msgs_0.7.5.bb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+SRC_URI[md5sum] = "60f559b7501006c2e021271ceef771b0"
+SRC_URI[sha256sum] = "170b74b99c54fcf0334df2c4b0ecef84007d3bc0be7f9c530e69edf391320b4c"
+
+S = "${WORKDIR}/${ROS_SP}"
+
+inherit catkin
+
+ROS_SPN = "moveit_msgs"
+
+DEPENDS = "message-generation std-msgs trajectory-msgs geometry-msgs sensor-msgs actionlib-msgs octomap-msgs object-recognition-msgs"
```
Comment (bulwahn) [inline recipes-ros/moveit-msgs/moveit-msgs_0.7.5.bb]: It is convention in openembedded that DEPENDS is put above the SRC_URI line. Move this line.

File: recipes-ros/moveit/moveit-ros-planning_0.7.8.bb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+DESCRIPTION = "Planning components of MoveIt that use ROS"
+SECTION = "devel"
+LICENSE = "BSD"
+LIC_FILES_CHKSUM = "file://package.xml;beginline=13;endline=13;md5=d566ef916e9dedc494f5f793a6690ba5"
+
+
```
Comment (bulwahn) [inline recipes-ros/moveit/moveit-ros-planning_0.7.8.bb]: Remove double empty line. One empty line is enough.

File: recipes-ros/moveit/moveit-ros-planning_0.7.8.bb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+require moveit-${PV}.inc
+
+ROS_BPN = "moveit_ros/planning"
+
+SRC_URI = "https://github.com/ros-planning/${ROS_SPN}/archive/${PV}.tar.gz;downloadfilename=${ROS_SP}.tar.gz \
+           file://0001-address-gcc6-build-error.patch \
+           "
+FILES_${PN} += "${ros_libdir}/moveit_ros_planning/*"
+
```
Comment (bulwahn) [inline recipes-ros/moveit/moveit-ros-planning_0.7.8.bb]: Drop the empty line at the end.

File: recipes-ros/srdfdom/srdfdom_0.3.5.bb [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+SRC_URI[md5sum] = "1b35fc7d120a8dbf41ede3c03c88ca5e"
+SRC_URI[sha256sum] = "9018c7c4da88e406e5059f0e2013049dfa4c04980f42a311c1201748ba5ba1de"
+
+S = "${WORKDIR}/${ROS_SP}"
+
+inherit catkin
+
+ROS_SPN = "srdfdom"
+
+DEPENDS = "cmake-modules boost urdf urdfdom-py console-bridge libtinyxml urdfdom-headers"
```
Comment (bulwahn) [inline recipes-ros/srdfdom/srdfdom_0.3.5.bb]: Put DEPENDS above SRC_URI.

**Review by bulwahn — CHANGES_REQUESTED** [review]
I quickly reviewed and it looks good. I just had some minor comments to some stylistic changes.

**Comment by bulwahn** [comment]
I also just tested `bitbake core-image-ros-world` with this build configuration:
```
Build Configuration:
BB_VERSION        = "1.33.2"
BUILD_SYS         = "x86_64-linux"
NATIVELSBSTRING   = "fedora-23"
TARGET_SYS        = "i586-oe-linux"
MACHINE           = "qemux86"
DISTRO            = "nodistro"
DISTRO_VERSION    = "nodistro.0"
TUNE_FEATURES     = "m32 i586"
TARGET_FPU        = ""
meta              = "HEAD:ac5c08ea00816852a712a8d7025bce7e865460ec"
meta-oe           
meta-python       
meta-multimedia   = "master:1dff2351aa6cdafa5a501e8956cb853ab17ed9ae"
meta-qt4          = "master:2c7f8df9039be498f8a2232d1428adb7f4e5e800"
meta-intel-realsense = "master:1d50cb5b82926f8e69a78999e9e8d2ffe2169897"
meta-ros          = "rojkov-moveit-v1:b4d98d9cdda14c1cc204b01f9ed9fba10338bead"
```

Everything builds successfully. Building fcl emits the following warning:
```
WARNING: fcl-0.3.2-r0 do_package_qa: QA Issue: fcl: /work/i586-oe-linux/fcl/0.3.2-r0/packages-split/fcl/usr/lib/libfcl.so contains probably-redundant RPATH /usr/lib [useless-rpaths]
```

This is not directly related to this pull request. `bitbake fcl` probably has this warning since quite some time; it is just become visible as it is now become part of the core-image-ros-world.

I will merge the pull request, no matter if the warning is addressed or not. I do not consider warnings as show stoppers for merging pull requests. If not addressed, I simply add an issue in our issue tracker to remember the issue and to let users know that we are aware.

**Comment by rojkov** [comment]
Thanks @bulwahn !

I have
* addressed the review comments;
* added a FCL patch fixing the RPATH issue;
* rebased and force pushed.

**Review by bulwahn — APPROVED** [review]
Looks good; I will run a quick test build and then merge this.

**Comment by bulwahn** [comment]
I tested the pull request successfully on these two build configurations:
```        
Build Configuration:
BB_VERSION        = "1.33.2"
BUILD_SYS         = "x86_64-linux"
NATIVELSBSTRING   = "fedora-23"
TARGET_SYS        = "i586-oe-linux"
MACHINE           = "qemux86"
DISTRO            = "nodistro"
DISTRO_VERSION    = "nodistro.0"
TUNE_FEATURES     = "m32 i586"
TARGET_FPU        = ""
meta              = "HEAD:ac5c08ea00816852a712a8d7025bce7e865460ec"
meta-oe           
meta-python       
meta-multimedia   = "master:1dff2351aa6cdafa5a501e8956cb853ab17ed9ae"
meta-qt4          = "master:2c7f8df9039be498f8a2232d1428adb7f4e5e800"
meta-intel-realsense = "master:1d50cb5b82926f8e69a78999e9e8d2ffe2169897"
meta-ros          = "rojkov-moveit-v2:cde0c0df72426a8f9f82caf6fc822093cd504949"
```

```
Build Configuration:
BB_VERSION        = "1.33.2"
BUILD_SYS         = "x86_64-linux"
NATIVELSBSTRING   = "ubuntu-14.04"
TARGET_SYS        = "i586-oe-linux"
MACHINE           = "qemux86"
DISTRO            = "nodistro"
DISTRO_VERSION    = "nodistro.0"
TUNE_FEATURES     = "m32 i586"
TARGET_FPU        = ""
meta              = "HEAD:ac5c08ea00816852a712a8d7025bce7e865460ec"
meta-oe           
meta-multimedia   
meta-python       = "master:6c584374a599f6f8d3607f20ecfc13a67ccf1da1"
meta-ros          = "rojkov-moveit-v2:cde0c0df72426a8f9f82caf6fc822093cd504949"
```

**Pull request merged**